features/cellular/framework/AT/AT_CellularSIM.cpp

```diff
@@ -56,9 +56,39 @@ nsapi_error_t AT_CellularSIM::get_sim_state(SimState &state)
         state = SimStateUnknown; // SIM may not be ready yet or +CPIN may be unsupported command
     }
     _at.resp_stop();
-    return _at.unlock_return_error();
+    _state = state;
+    nsapi_error_t error = _at.get_last_error();
+    _at.unlock();
+    trace_sim_errors();    
+    return error;
 }
 
+
+CellularSIM::SimState AT_CellularSIM::trace_sim_errors(void)
```

Comment:
you could add this inside #if MBED_CONF_MBED_TRACE_ENABLE so when trace not enabled no extra code is generated